diff --git a/arch/arm/boot/dts/zero-sugar.dts b/arch/arm/boot/dts/zero-sugar.dts
index 8aa62bc2c468..aa7327d47e18 100644
--- a/arch/arm/boot/dts/zero-sugar.dts
+++ b/arch/arm/boot/dts/zero-sugar.dts
@@ -50,7 +50,7 @@
 	compatible = "fsl,imx7d-sdb", "fsl,imx7d";
 
 	chosen {
-		stdout-path = &uart1;
+		stdout-path = &uart6;
 	};
 
 	aliases {
@@ -725,6 +725,8 @@
 };
 
 &uart6 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart6>;
 	assigned-clocks = <&clks IMX7D_UART6_ROOT_SRC>;
 	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
 	status = "okay";
@@ -926,6 +928,13 @@
 		>;
 	};
 
+	pinctrl_uart6: uart6grp {
+		fsl,pins = <
+			MX7D_PAD_EPDC_DATA09__UART6_DCE_TX	0x79
+			MX7D_PAD_EPDC_DATA08__UART6_DCE_RX	0x79
+		>;
+	};
+
 	pinctrl_usdhc2: usdhc2grp {
 		fsl,pins = <
 			MX7D_PAD_SD2_CMD__SD2_CMD		0x59
@@ -1055,13 +1064,13 @@
 	pinctrl_one_wire_uart6_tx: one_wire_uart6_tx_grp {
 		fsl,pins = <
 			MX7D_PAD_EPDC_DATA08__GPIO2_IO8		0x00000004
-			MX7D_PAD_EPDC_DATA09__UART6_DCE_TX	0x00000004
+			MX7D_PAD_EPDC_DATA09__UART6_DCE_TX	0x00000079
 		>;
 	};
 
 	pinctrl_one_wire_uart6_rx: one_wire_uart6_rx_grp {
 		fsl,pins = <
-			MX7D_PAD_EPDC_DATA08__UART6_DCE_RX	0x00000004
+			MX7D_PAD_EPDC_DATA08__UART6_DCE_RX	0x00000079
 			MX7D_PAD_EPDC_DATA09__GPIO2_IO9		0x00000004
 		>;
 	};
